packages/server-wallet/src/protocols/direct-funder.ts

```diff
@@ -0,0 +1,120 @@
+import {Address, BN, checkThat, isSimpleAllocation, Uint256} from '@statechannels/wallet-core';
+import _ from 'lodash';
+import {Transaction} from 'objection';
+import {Logger} from 'pino';
+
+import {ChainServiceInterface} from '../chain-service';
+import {ChainServiceRequest} from '../models/chain-service-request';
+import {Channel} from '../models/channel';
+import {DBOpenChannelObjective} from '../models/objective';
+import {Store} from '../wallet/store';
+import {WalletResponse} from '../wallet/wallet-response';
+
+export class DirectFunder {
+  constructor(
+    private store: Store,
+    private chainService: ChainServiceInterface,
+    private logger: Logger,
+    private timingMetrics = false
+  ) {}
+
+  public static create(
+    store: Store,
+    chainService: ChainServiceInterface,
+    logger: Logger,
+    timingMetrics = false
+  ): DirectFunder {
+    return new DirectFunder(store, chainService, logger, timingMetrics);
+  }
+
+  /**
+   * Runs the ledger funding protocol
+   *
+   * Returns true if the channel is funded, false otherwise
```

Comment:
I'm not sure. The `Funders` have a fundamentally different interface, as their `crank` methods accept a transaction, whereas the `Opener` creates its own transaction.

I think this will become clearer over time, when we have more of them.